src/main/java/it/reply/orchestrator/service/deployment/providers/MarathonServiceImpl.java

```diff
@@ -327,18 +328,24 @@ public boolean doUndeploy(DeploymentMessage deploymentMessage) {
       }
     }
 
-    TokenAuthentication vaultToken = vaultService.retrieveToken(requestedWithToken);
+    try {
+      TokenAuthentication vaultToken = vaultService.retrieveToken(requestedWithToken);
 
-    //remove vault entries if present
-    String spath = "secret/private/" + deployment.getId();
-    List<String> depentries = vaultService.listSecrets(vaultToken, spath);
-    if (!depentries.isEmpty()) {
-      for (String depentry:depentries) {
-        List<String> entries = vaultService.listSecrets(vaultToken, spath + "/" + depentry);
-        for (String entry:entries) {
-          vaultService.deleteSecret(vaultToken, spath + "/" + depentry + "/" + entry);
+      //remove vault entries if present
+      String spath = "secret/private/" + deployment.getId();
+      List<String> depentries = vaultService.listSecrets(vaultToken, spath);
+      if (!depentries.isEmpty()) {
+        for (String depentry:depentries) {
+          List<String> entries = vaultService.listSecrets(vaultToken, spath + "/" + depentry);
+          for (String entry:entries) {
+            vaultService.deleteSecret(vaultToken, spath + "/" + depentry + "/" + entry);
+          }
         }
       }
+    } catch (VaultServiceNotAvailableException ex) {
+      // skip gracefully if vault not implemented
```

Comment:
In order for this situation to occur, the configuration of the orchestrator with respect to the time of deployment should be changed; this could happen because, for example, the Vault server has been changed (and in this case the Secrets have been lost), that is, the service is no longer available.
The Secrets are by nature (the reason for their use) not easily accessible to a person external to the system, moreover, they are indexed using the deployment uuid that is being canceled and therefore their use could hardly be traced.
It does not seem to me, honestly, a situation that is likely to occur; the handling of the exception has been inserted for the sole purpose of not letting the deletion of the deployment completely fail. In any case, if you still want to inform the user about the persistence in an undefined server of some of his Secret, suggest me a way to make the message bubble-up.